source/main.c

```diff
@@ -208,6 +215,27 @@ int main()
     u32 t = ticks_to_secs(SYS_Time());
     settime(secs_to_ticks(t));
 
+    PAD_ScanPads();
+    
+    all_buttons_held = (
+        PAD_ButtonsHeld(PAD_CHAN0) |
+        PAD_ButtonsHeld(PAD_CHAN1) |
+        PAD_ButtonsHeld(PAD_CHAN2) |
+        PAD_ButtonsHeld(PAD_CHAN3)
+    );
+    
+         if (all_buttons_held & PAD_BUTTON_A    ) strcpy(path, "/a.dol"    );
+    else if (all_buttons_held & PAD_BUTTON_B    ) strcpy(path, "/b.dol"    );
+    else if (all_buttons_held & PAD_BUTTON_X    ) strcpy(path, "/x.dol"    );
+    else if (all_buttons_held & PAD_BUTTON_Y    ) strcpy(path, "/y.dol"    );
+    else if (all_buttons_held & PAD_TRIGGER_Z   ) strcpy(path, "/z.dol"    );
+    else if (all_buttons_held & PAD_BUTTON_START) strcpy(path, "/start.dol");
+    else if (all_buttons_held & PAD_BUTTON_LEFT ) strcpy(path, "/left.dol" );
+    else if (all_buttons_held & PAD_BUTTON_RIGHT) strcpy(path, "/right.dol");
+    else if (all_buttons_held & PAD_BUTTON_UP   ) strcpy(path, "/up.dol"   );
```

Comment:
This is very messy. There is no reason to use `strcpy` here. Make a table of paths and use an offset or pointer to select the correct one.
Bonus points for getting rid of the `else if` chain (I'd make a table of bit masks and loop over it).

Reply:
Can remove `strcpy`. Again was just going off seewood's commit and C ignorance.

Funny, I think this is much more readable than defining a table and looping over it. Very clear intent.

You are proposing something like this?

```c
struct shortcut {
  u16 padButtons;
  char* path;
} shortcuts[] = {
  {PAD_BUTTON_A,     "/a.dol"    },
  {PAD_BUTTON_B,     "/b.dol"    },
  {PAD_BUTTON_X,     "/x.dol"    },
  {PAD_BUTTON_Y,     "/y.dol"    },
  {PAD_TRIGGER_Z,    "/z.dol"    },
  {PAD_BUTTON_START, "/start.dol"},
  {PAD_BUTTON_LEFT,  "/left.dol" },
  {PAD_BUTTON_RIGHT, "/right.dol"},
  {PAD_BUTTON_UP,    "/up.dol"   }
}
int numShortcuts = sizeof(shortcuts)/sizeof(shortcuts[0]);

// ...

for (int i = 0; i < numShortcuts; i++) {
  if (all_buttons_held & shortcuts[i].padButtons) {
    path = shortcuts[i].path;
    break;
  }
}
```

Reply:
Yes, that looks pretty good. Just change camelCase to snake_case please.

Reply:
You raise a fair point with regards to expressing intent. I don't think my proposed solution is worse, I could ask someone else to weigh in on this. It would also be interesting to throw this into Compiler Explorer and see what they compile to. My guess is that the loop would result in slightly smaller code size, and ideally we should try to optimize for that.

Reply:
I also do not feel strongly either way. Just personal preference on readability. I'd also be curious how each is compiled. My intuition thinks branches would be easier to optimize, but I don't know much of anything about the compiler.